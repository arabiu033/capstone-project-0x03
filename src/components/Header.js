import PropTypes from 'prop-types';
import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = (props) => {
  const { title } = props;

  return (
    <div style={{ display: 'flex', justifyContent: 'space-around' }}>
      <span><NavLink to="/" onClick={() => title[1](() => 'Categories')}>Home</NavLink></span>
      <span>{ title[0] }</span>
      <ul>
        <span>Mic</span>
        <span>Settings</span>
      </ul>
    </div>
  );
};

Header.propTypes = {
  title: PropTypes.arrayOf(PropTypes.elementType).isRequired,
};
export default Header;
