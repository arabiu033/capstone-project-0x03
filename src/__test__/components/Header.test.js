import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/configureStore';

import Header from '../../components/Header';

test('renders correctly Form component', () => {
  const header = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <Header title={[]}/>
      </Provider>
    </BrowserRouter>
  ).toJSON();
  expect(header).toMatchSnapshot();
});