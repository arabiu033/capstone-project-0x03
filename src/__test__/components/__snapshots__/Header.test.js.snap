// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Form component 1`] = `
<div
  className="header-container"
>
  <span>
    <a
      aria-current="page"
      className="header-home dhead active"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </span>
  <span
    className="header-title dhead"
  />
  <ul>
    <img
      alt="mic"
      className="header-icons"
      src="mic.png"
    />
    <img
      alt="setting"
      className="header-icons"
      src="setting.png"
    />
  </ul>
</div>
`;
